packages/wrangler/src/dev.tsx

```diff
@@ -533,6 +534,251 @@ export async function startDev(args: StartDevOptions) {
 	}
 }
 
+export async function startApiDev(args: StartDevOptions) {
+	let watcher: ReturnType<typeof watch> | undefined;
+
+	try {
+		if (args.logLevel) {
+			// we don't define a "none" logLevel, so "error" will do for now.
+			logger.loggerLevel = args.logLevel === "none" ? "error" : args.logLevel;
+		}
+		await printWranglerBanner();
+
+		const configPath =
+			(args.config as ConfigPath) ||
+			((args.script &&
+				findWranglerToml(path.dirname(args.script))) as ConfigPath);
+		let config = readConfig(configPath, args);
+		await metrics.sendMetricsEvent(
+			"run dev",
+			{ local: args.local },
+			{ sendMetrics: config.send_metrics, offline: args.local }
+		);
+
+		if (config.configPath) {
+			watcher = watch(config.configPath, {
+				persistent: true,
+			}).on("change", async (_event) => {
+				// TODO: Do we need to handle different `_event` types differently?
+				//       e.g. what if the file is deleted, or added?
+				config = readConfig(configPath, args);
+				if (config.configPath) {
+					logger.log(`${path.basename(config.configPath)} changed...`);
+					//TODO: @rozenmd: re-run generateBundle etc here
+					// rerender(await getDevReactElement(config));
+				}
+			});
+		}
+
+		const entry = await getEntry(
+			{ assets: args.assets, script: args.script },
+			config,
+			"dev"
+		);
+
+		if (config.services && config.services.length > 0) {
```

Comment:
I feel like all of these logger.warn/errors should be common, right @threepointone ?